app/controllers/rescue_requests_controller.rb

```diff
@@ -71,6 +72,15 @@ def apply_triage_status
     redirect_to disaster_request_path(disaster_id: @disaster.id, num: @request.incident_number)
   end
 
+  def apply_medical_triage_status
+    if current_user.has_role?(:medical) && @request.update(medical_status_id: params[:status_id])
```

Comment:
Use a before_action callback to check roles, and use `has_any_role? :developer, :medical' in it